application/classes/Ushahidi/Repository/Message.php

```diff
@@ -189,6 +189,23 @@ public function checkStatus($status, $direction)
 		return false;
 	}
 
+	public function getLastUID($data_provider_type)
+	{
+		$last_uid = null;
+		$query = DB::select([DB::expr('ABS(' . $this->getTable() . '.' . 'data_provider_message_id' . ')'), 'uid'])
+			->from($this->getTable())
+			->where('data_provider', '=', $data_provider_type)
+			->order_by(
```

Comment:
Can you add a limit clause to this? unless I'm missing something we're loading *all* messages just to get the lastest one?